daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/Ast.scala

```diff
@@ -504,6 +504,12 @@ object Ast {
       choice: ChoiceName,
       cidE: Expr,
       argE: Expr,
+      guardE: Option[Expr],
+      // `guardE` is an optional expression of type Interface -> Bool which is evaluated
+      // after fetching the contract but before running the exercise body. If the guard
+      // returns false, the transaction is aborted. If the guard raises an exception,
```

Comment:
maybe worth stating here that the fetch nodes are not yet inserted.

Reply:
added a TODO

Reply:
removed the TODO and updated the comment, since we're going to abort in every case